src/NUnitFramework/tests/Attributes/TestCaseSourceTests.cs

```diff
@@ -290,6 +290,21 @@ public void HandlesExceptionInTestCaseSource_GuiDisplay(string lhs, string rhs)
             Assert.AreEqual(lhs, rhs);
         }
 
+
+        private static IEnumerable<TestCaseData> ZeroTestCasesSource() => Enumerable.Empty<TestCaseData>();
+
+        [TestCaseSource(nameof(ZeroTestCasesSource))]
+        public void Test_with_zero_test_cases_should_not_be_run()
```

Comment:
No, good comment and I did it intentionally for the POC. @CharliePoole will probably agree with you?